src/Npgsql/NpgsqlTypes/NpgsqlDate.cs

```diff
@@ -1,452 +1,235 @@
-            }
-            var maxDay = (IsLeap(newYear) ? LeapYearMaxes : CommonYearMaxes)[newMonth - 1];
-            var newDay = Day > maxDay ? maxDay : Day;
-            return new NpgsqlDate(newYear, newMonth, newDay);
-
-        }
-
-        public NpgsqlDate Add(in NpgsqlTimeSpan interval)
-        {
-            switch (_type) {
-            case InternalType.Infinity:
-                return Infinity;
-            case InternalType.NegativeInfinity:
-                return NegativeInfinity;
-            case InternalType.Finite:
-                break;
-            default:
-                throw new InvalidOperationException($"Internal Npgsql bug: unexpected value {_type} of enum {nameof(NpgsqlDate)}.{nameof(InternalType)}. Please file a bug.");
-            }
-
-            return AddMonths(interval.Months).AddDays(interval.Days);
-        }
-
-        internal NpgsqlDate Add(in NpgsqlTimeSpan interval, int carriedOverflow)
-        {
-            switch (_type) {
-            case InternalType.Infinity:
-                return Infinity;
-            case InternalType.NegativeInfinity:
-                return NegativeInfinity;
-            case InternalType.Finite:
-                break;
-            default:
-                throw new InvalidOperationException($"Internal Npgsql bug: unexpected value {_type} of enum {nameof(NpgsqlDate)}.{nameof(InternalType)}. Please file a bug.");
-            }
-
-            return AddMonths(interval.Months).AddDays(interval.Days + carriedOverflow);
-        }
-
-        #endregion
-
-        #region Comparison
-
-        public int Compare(NpgsqlDate x, NpgsqlDate y) => x.CompareTo(y);
-
-        public int Compare(object? x, object? y)
-        {
-            if (x == null)
-            {
-                return y == null ? 0 : -1;
-            }
-            if (y == null)
-            {
-                return 1;
-            }
-            if (!(x is IComparable) || !(y is IComparable))
-            {
-                throw new ArgumentException();
-            }
-            return ((IComparable) x).CompareTo(y);
-        }
-
-        public bool Equals(NpgsqlDate other)
-            => _type switch
-            {
-                InternalType.Infinity         => other._type == InternalType.Infinity,
-                InternalType.NegativeInfinity => other._type == InternalType.NegativeInfinity,
-                InternalType.Finite           => other._type == InternalType.Finite && _daysSinceEra == other._daysSinceEra,
-                _ => false
-            };
-
-        public override bool Equals(object? obj) => obj is NpgsqlDate date && Equals(date);
+        /// <summary>
+        /// The value of this constant is equivalent to January 1, 2000, in the Gregorian calendar.
+        /// <see cref="PostgreSqlEpoch"/> defines the point in time when PostgreSQL time is equal to 0.
+        /// </summary>
+        public static NpgsqlDate PostgreSqlEpoch => new NpgsqlDate(0);
```

Comment:
As above, I don't think it makes sense to expose this to users - it's an internal PG concern.

Reply:
Since PostgreSQL doesn't expose it publicly too, I agree on it, but then that part should be mentioned in the docs stating that Jan 1 of 2000 is a default value for `NpgsqlDate` and others.